Fixes #593, others - issues with pyilmbase install

kdt3rd opened the pull request:
The PyIlmBase install logic was incomplete in cmake. This fixes this:
- computes a relative path from the python paths that can then be used
with CMAKE_INSTALL_PREFIX to install python modules to an appropriate
location (can be manually configured as well, install docs updated)
- Fixes #604, pyilmbaseconfig was also not being installed properly

Signed-off-by: Kimball Thurston <kdt3rd@gmail.com>

kdt3rd committed:
Fixes #593, others - issues with pyilmbase install

The PyIlmBase install logic was incomplete in cmake. This fixes this:
- computes a relative path from the python paths that can then be used
with CMAKE_INSTALL_PREFIX to install python modules to an appropriate
location (can be manually configured as well, install docs updated)
- Fixes #604, pyilmbaseconfig was also not being installed properly

Signed-off-by: Kimball Thurston <kdt3rd@gmail.com>

meshula reviewed (approved):
nice

kdt3rd committed:
Extract to function, protect against infinite loop

Seems like under windows, the introduced relative path search can cause
an infinite loop if it doesn't match. Test for that, may need
adjustment.

Signed-off-by: Kimball Thurston <kdt3rd@gmail.com>

kdt3rd committed:
Fix issue with drive letter under windows

The get_filename_component gets stuck under windows when finding the
parent directory for a drive letter. This basically reverts to using the
full path, which is more likely to be ok under windows

Signed-off-by: Kimball Thurston <kdt3rd@gmail.com>

kdt3rd committed:
ensure paths are canonicalized by get_filename_component prior to comparing

This fixes an issue under windows where the root is not properly
strequal

Signed-off-by: Kimball Thurston <kdt3rd@gmail.com>

meshula commented on PyIlmBase/CMakeLists.txt:
```diff
@@ -55,48 +55,47 @@ endif()
 # for our other dependency - boost.
 # Boost Python has some .. annoyances in that the python module
 # has version names attached to it
+function(PYILMBASE_EXTRACT_REL_SITEARCH varname pyver pyexe pysitearch)
```
I bet it's going to be far less work to port Imath to pybind11, than to deal with endless boost::python/cmake curveballs!

meshula commented on PyIlmBase/CMakeLists.txt:
```diff
@@ -106,7 +105,6 @@ endif()
 # let's run a search and see what we get instead of making it
 # an explicit required. The older names are not portable, but
 # we'll do the best we can
```
Boost_NO_BOOST_CMAKE protects the build from the FindBoost.cmake from failing unnecessarily, as I think was the solution for one of the MinGW issue threads. Will the advice be to set it from the command line if one is in a sufficiently off-the-beaten-path boost environment? Is there any reason not to retain Boost_NO_BOOST_CMAKE here?

kdt3rd replied:
Yes, I wanted to test with and without it. I am going to leave it on (about to restore it), as I think it fixes more than it breaks. I'm also adding a note in the INSTALL.md in the prerequisites section and in the error message I print out about this in the CMakeLists.txt hoping it helps people find that we are trying and maybe give them a clue to look for

kdt3rd committed:
Set CMP0074 such that people who set Boost_ROOT won't get warnings

Signed-off-by: Kimball Thurston <kdt3rd@gmail.com>

kdt3rd committed:
Re-enable Boost_NO_BOOST_CMAKE by default, document, clean up status messages

Signed-off-by: Kimball Thurston <kdt3rd@gmail.com>

kdt3rd committed:
Fix issue with defines not being set correctly for win32

we always build shared libraries now, so even if the default is to build
static, the python bindings are forced to build shared, which means they
need to set the defines to export the symbols appropriately, add extra
variable set

Signed-off-by: Kimball Thurston <kdt3rd@gmail.com>

kdt3rd committed:
Handle python2 not being installed, but python3 being present

Signed-off-by: Kimball Thurston <kdt3rd@gmail.com>

kdt3rd committed:
Fix overzealous removal of if statements breaking all builds except win32

Signed-off-by: Kimball Thurston <kdt3rd@gmail.com>
